financial-connections/src/main/java/com/stripe/android/financialconnections/domain/SaveAccountToLink.kt

```diff
@@ -19,38 +24,86 @@ internal class SaveAccountToLink @Inject constructor(
         email: String,
         phoneNumber: String,
-        selectedAccounts: Set<String>,
-        country: String
-    ): FinancialConnectionsSessionManifest = runCatching {
-        repository.postSaveAccountsToLink(
-            clientSecret = configuration.financialConnectionsSessionClientSecret,
-            email = email,
-            country = country,
-            phoneNumber = phoneNumber,
-            locale = (locale ?: Locale.getDefault()).toLanguageTag(),
-            consumerSessionClientSecret = null,
-            selectedAccounts = selectedAccounts
-        )
+        selectedAccounts: List<PartnerAccount>,
+        country: String,
+        shouldPollAccountNumbers: Boolean,
+    ): FinancialConnectionsSessionManifest {
+        return withReadyAccounts(shouldPollAccountNumbers, selectedAccounts) { selectedAccountIds ->
+            repository.postSaveAccountsToLink(
+                clientSecret = configuration.financialConnectionsSessionClientSecret,
+                email = email,
+                country = country,
+                phoneNumber = phoneNumber,
+                locale = (locale ?: Locale.getDefault()).toLanguageTag(),
+                consumerSessionClientSecret = null,
+                selectedAccounts = selectedAccountIds,
+            )
+        }
     }
-        .updateCustomSuccessMessage(selectedAccounts.size)
-        .getOrThrow()
 
     suspend fun existing(
         consumerSessionClientSecret: String,
-        selectedAccounts: Set<String>,
-    ): FinancialConnectionsSessionManifest = runCatching {
-        repository.postSaveAccountsToLink(
+        selectedAccounts: List<PartnerAccount>,
+        shouldPollAccountNumbers: Boolean,
+    ): FinancialConnectionsSessionManifest {
+        return withReadyAccounts(shouldPollAccountNumbers, selectedAccounts) { selectedAccountIds ->
+            repository.postSaveAccountsToLink(
+                clientSecret = configuration.financialConnectionsSessionClientSecret,
+                email = null,
+                country = null,
+                phoneNumber = null,
+                locale = null,
+                consumerSessionClientSecret = consumerSessionClientSecret,
+                selectedAccounts = selectedAccountIds,
+            )
+        }
+    }
+
+    private suspend fun withReadyAccounts(
+        shouldPollAccountNumbers: Boolean,
+        partnerAccounts: List<PartnerAccount>,
+        action: suspend (Set<String>) -> FinancialConnectionsSessionManifest,
+    ): FinancialConnectionsSessionManifest {
+        val selectedAccountIds = partnerAccounts.map { it.id }.toSet()
+        val linkedAccountIds = partnerAccounts.mapNotNull { it.linkedAccountId }.toSet()
+
+        val pollingResult = if (shouldPollAccountNumbers) {
+            pollAccountNumbers(linkedAccountIds).onFailure {
+                storeFailedToSaveToLinkMessage(selectedAccounts = partnerAccounts.size)
+                disableNetworking()
+            }
```

Comment:
We disable networking and set the appropriate `customSuccessMessage` if polling fails.